packages/client/src/client.ts

```diff
@@ -272,10 +272,10 @@ export class SkipClient {
       gasAmountMultiplier = DEFAULT_GAS_MULTIPLIER,
       getFallbackGasAmount,
     } = options;
-    let gasTokenUsed: Coin | undefined;
+    let gasTokensUsed: Record<string, Coin> | undefined;
     if (validateGasBalance) {
       // check balances on chains where a tx is initiated
-      gasTokenUsed = await this.validateGasBalances({
+      gasTokensUsed = await this.validateGasBalances({
```

Comment:
does `this.validateGasBalances` return a record<string, Coin> of gasTokenUsed?

Reply:
yes

Reply:
@codingki it's a little confusing having `gasTokenUsed` and `gasTokensUsed`, maybe change one of them to be called GasTokenArray? also is gasTokenUsed an array or record? `gasTokensUsed?.[i]` being used in a loop suggests its an array

Reply:
it's a record. we can use a record like that